### PR TITLE
feat: introduce Json converter for SurfaceBounds

### PR DESCRIPTION
**Pull request by asalzburger**
This PR includes the next `Json` converter, this time for `SurfaceBounds`.

All `SurfaceBounds` have a `values()` streaming method (providing a `std::vector`) and a constructor by providing a `std::array`. Hence, only one converter is needed, however, the type has to be given explicitely, no automated conversion is possible, as the `Surface` objects never swallow `SurfaceBounds` directly but a class extension of the appropriate type.

UnitTest for one is attached, no need to call the same functions on all inherited objects.

### CODE REVIEW

File: Core/include/Acts/Surfaces/SurfaceBounds.hpp [hunk 1]
```diff
@@ -33,16 +33,16 @@ class SurfaceBounds {
...
+    eLine = 5,
+    eRectangle = 6,
+    eTrapezoid = 7,
+    eTriangle = 8,
+    eDiscTrapezoid = 9,
+    eConvexPolygon = 10,
+    eAnnulus = 11,
+    eBoundless = 12,
+    eOther = 13
   };
```
Comment (asalzburger) [inline Core/include/Acts/Surfaces/SurfaceBounds.hpp:46]: There was a counting bug.

File: Plugins/Json/src/UtilitiesJsonConverter.cpp [hunk 2]
```diff
@@ -77,9 +77,7 @@ void Acts::from_json(const nlohmann::json& j, Acts::BinningData& bd) {
 void Acts::to_json(nlohmann::json& j, const Acts::BinUtility& bu) {
   nlohmann::json jbindata;
   for (const auto& bdata : bu.binningData()) {
-    nlohmann::json jdata;
-    to_json(jdata, bdata);
-    jbindata.push_back(jdata);
+    jbindata.push_back(nlohmann::json(bdata));
```
Comment (asalzburger) [inline Plugins/Json/src/UtilitiesJsonConverter.cpp:80]: Trying out the new auto-conversion feature.

**Comment by asalzburger** [comment]
As @paulgessinger suggested: probably put the `Transform3` into `Eigen` namespace.

File: Plugins/Json/src/UtilitiesJsonConverter.cpp (hunk 2 above)
Comment (Corentin-Allaire) [reply Plugins/Json/src/UtilitiesJsonConverter.cpp:80]: That is also what I am using in my PR.

Comment (Corentin-Allaire) [reply Plugins/Json/src/UtilitiesJsonConverter.cpp:80]: Note than you can also use it the other way around, for example line 98 you could write: 
Acts::BinningData bd(jdata);

**Review by Corentin-Allaire — APPROVED** [review]
I approve. I saw that you started using the json auto-conversion in Plugins/Json/src/UtilitiesJsonConverter.cpp do you want to replace all the to_json and from_json in that file ?

File: Plugins/Json/src/UtilitiesJsonConverter.cpp (hunk 2 above)
Comment (asalzburger) [reply Plugins/Json/src/UtilitiesJsonConverter.cpp:80]: Ok, I will clean out with the next PR and save a few lines.

**Pull request merged**